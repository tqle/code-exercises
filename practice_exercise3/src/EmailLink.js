import React from "react";
import { validateEmail } from "./utils/linkUtil";

/**
 * Email Link extends link tag with
 * custom mailto: href to indicate link is an email.
 */
class EmailLink extends React.Component {
  render() {
    // Destructuring objects from props.
    let { email } = this.props;

    // Check for empty, undefined, null cases
    let validEmail = validateEmail(email);
    // FUTURE: Write a regex or pattern match if email is valid format
    let title = validEmail || "";
    let linkReference = validEmail ? "mailto:" + validEmail : "#";

    return (
      <a title={title} href={linkReference}>
        {validEmail}
      </a>
    );
  }
}

export default EmailLink;
