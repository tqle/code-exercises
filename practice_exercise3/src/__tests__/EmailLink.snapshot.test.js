import React from "react";
import { create } from "react-test-renderer";
import EmailLink from "../EmailLink";

/**
 * Test Email Link snapshot
 */
test("<Email /> renders correctly", () => {
  const email = "Julianne.OConner@kory.org";
  const test = create(<EmailLink email={email} />);
  expect(test.toJSON()).toMatchSnapshot();
});

test("<Email /> renders as an anchor when no email reference is set", () => {
  const email = "Julianne.OConner@kory.org";
  const test = create(<EmailLink>{email}</EmailLink>).toJSON();
  expect(test).toMatchSnapshot();
});
