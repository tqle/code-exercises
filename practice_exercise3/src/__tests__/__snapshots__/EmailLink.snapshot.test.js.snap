// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Email /> renders as an anchor when no email reference is set 1`] = `
<a
  href="#"
  title=""
>
  
</a>
`;

exports[`<Email /> renders correctly 1`] = `
<a
  href="mailto:Julianne.OConner@kory.org"
  title="Julianne.OConner@kory.org"
>
  Julianne.OConner@kory.org
</a>
`;
